presto-cassandra/src/main/java/com/facebook/presto/cassandra/NativeCassandraSession.java

```diff
@@ -253,6 +268,33 @@ public CassandraTable getTable(SchemaTableName schemaTableName)
 
     private KeyspaceMetadata getKeyspaceByCaseInsensitiveName(String caseInsensitiveSchemaName)
             throws SchemaNotFoundException
+    {
+        try {
+            return keyspaceCache.get(caseInsensitiveSchemaName);
+        }
+        catch (Exception e) {
+            if (e.getCause() != null) {
+                Throwable cause = e.getCause();
+                if (cause instanceof SchemaNotFoundException) {
+                    throw (SchemaNotFoundException) cause;
+                }
+
+                if (cause instanceof PrestoException) {
+                    throw (PrestoException) cause;
+                }
+            }
+
+            if (RuntimeException.class.isAssignableFrom(e.getClass())) {
+                throw (RuntimeException) e;
+            }
+            else {
+                throw new RuntimeException(e);
+            }
+        }
```

Comment:
Seems like we can just catch `ExecutionException` for processing, and let subclasses of `RuntimeException` throw?

```suggestion
        catch (ExecutionException e) {
            Throwable cause = e.getCause();
            if (cause instanceof SchemaNotFoundException) {
                throw (SchemaNotFoundException) cause;
            }

            if (cause instanceof PrestoException) {
                throw (PrestoException) cause;
            }
        }
```